grr/server/grr_response_server/gui/ui/components/flow_details/helpers/osquery_test_utils.ts

```diff
@@ -0,0 +1,208 @@
+import { OsqueryColumn, OsqueryRow } from '@app/lib/api/api_interfaces';
+import { newFlowListEntry, newFlowResultSet } from '@app/lib/models/model_test_util';
+import { DeepMutable } from '@app/lib/type_utils';
+import { FlowListEntry, FlowState, FlowResultSet } from '@app/lib/models/flow';
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+
+type FullMutableOsqueryTable = {
+  query: string,
+  header: {
+    columns: ReadonlyArray<OsqueryColumn>,
+  },
+  rows: ReadonlyArray<OsqueryRow>
+};
+
+function newOsqueryTable(): FullMutableOsqueryTable {
+  return {
+    query: '',
+    header: {
+      columns: [] as ReadonlyArray<OsqueryColumn>,
```

Comment:
Why explicit casting to ReadonlyArray is necessary here?

Reply:
`[]` has inferred type `never[]`, because as far as Typescript is concerned, it is always an empty array.